Compiler Crash (NPE) when using defaults in super call and missing implicit

```scala
class C(x: Int = 0, y: Int = 0)(implicit blup: String)
class D extends C(y = 1)
```

The compiler reports the missing implicit and then crashes

```
/Users/luc/code/scala/scala13/sandbox/T.scala:2: error: could not find implicit value for parameter blup: String
Error occurred in an application involving default arguments.
class D extends C(y = 1)
                ^
error: java.lang.NullPointerException: Cannot invoke "scala.reflect.internal.Symbols$Symbol.isPrimaryConstructor()" because the return value of "scala.reflect.internal.Trees$Tree.symbol()" is null
	at scala.tools.nsc.typechecker.Typers$Typer.analyzeSuperConstructor(Typers.scala:2292)
	at scala.tools.nsc.typechecker.Typers$Typer.typedDefDef(Typers.scala:2482)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:6167)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:6261)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:6339)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$9(Typers.scala:3539)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3539)
	at scala.tools.nsc.typechecker.Typers$Typer.typedTemplate(Typers.scala:2144)
	at scala.tools.nsc.typechecker.Typers$Typer.typedClassDef(Typers.scala:1982)
```

--- a/src/compiler/scala/tools/nsc/typechecker/Typers.scala
+++ b/src/compiler/scala/tools/nsc/typechecker/Typers.scala
@@ -2247,15 +2247,13 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
     }
 
     /** Analyze the super constructor call to record information used later to compute parameter aliases */
-    def analyzeSuperConstructor(meth: Symbol, vparamss: List[List[ValDef]], rhs: Tree): Unit = {
+    def analyzeSuperConstructor(meth: Symbol, vparamss: List[List[ValDef]], rhs: Tree): Unit = if (!rhs.isErrorTyped) {
       val clazz = meth.owner
       debuglog(s"computing param aliases for $clazz:${clazz.primaryConstructor.tpe}:$rhs")
       val pending = ListBuffer[AbsTypeError]()
 
       // !!! This method is redundant with other, less buggy ones.
       def decompose(call: Tree): (Tree, List[Tree]) = call match {
-        case _ if call.isErrorTyped => // e.g. scala/bug#7636
-          (call, Nil)
         case Apply(fn, args) =>
           // an object cannot be allowed to pass a reference to itself to a superconstructor
           // because of initialization issues; scala/bug#473, scala/bug#3913, scala/bug#6928.

--- a/test/files/neg/t13014.check
+++ b/test/files/neg/t13014.check
@@ -1,0 +1,5 @@
+t13014.scala:2: error: could not find implicit value for parameter blup: String
+Error occurred in an application involving default arguments.
+class D extends C(y = 1)
+                ^
+1 error

--- a/test/files/neg/t13014.scala
+++ b/test/files/neg/t13014.scala
@@ -1,0 +1,2 @@
+class C(x: Int = 0, y: Int = 0)(implicit blup: String)
+class D extends C(y = 1)

